internal/store/pod.go

```diff
@@ -728,6 +728,73 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_pod_resources",
+			Type: metric.Gauge,
+			Help: "The resource requests and limits for a pod.",
+			GenerateFunc: wrapPodFunc(func(p *v1.Pod) *metric.Family {
+				ms := []*metric.Metric{}
+
+				lifecycle, reqs, currentReqs, limits, currentLimits := podRequestsAndLimitsByLifecycle(p)
+				for _, t := range []struct {
+					lifecycle string
+					name      string
+					resources v1.ResourceList
+				}{
+					{lifecycle: "", name: "requests", resources: currentReqs},
+					{lifecycle: lifecycle, name: "requests", resources: reqs},
+					{lifecycle: "", name: "limits", resources: currentLimits},
+					{lifecycle: lifecycle, name: "limits", resources: limits},
+				} {
+					req := t.resources
+					for resourceName, val := range req {
+						switch resourceName {
+						case v1.ResourceCPU:
+							ms = append(ms, &metric.Metric{
+								LabelValues: []string{t.lifecycle, "", t.name, sanitizeLabelName(p.Spec.NodeName), sanitizeLabelName(string(resourceName)), string(constant.UnitCore)},
+								Value:       float64(val.MilliValue()) / 1000,
+							})
+						case v1.ResourceStorage:
+							fallthrough
+						case v1.ResourceEphemeralStorage:
+							fallthrough
+						case v1.ResourceMemory:
+							ms = append(ms, &metric.Metric{
+								LabelValues: []string{t.lifecycle, "", t.name, sanitizeLabelName(p.Spec.NodeName), sanitizeLabelName(string(resourceName)), string(constant.UnitByte)},
+								Value:       float64(val.Value()),
+							})
+						default:
+							if isHugePageResourceName(resourceName) {
+								ms = append(ms, &metric.Metric{
+									LabelValues: []string{t.lifecycle, "", t.name, p.Spec.NodeName, sanitizeLabelName(string(resourceName)), string(constant.UnitByte)},
+									Value:       float64(val.Value()),
+								})
+							}
+							if isAttachableVolumeResourceName(resourceName) {
+								ms = append(ms, &metric.Metric{
+									LabelValues: []string{t.lifecycle, "", t.name, p.Spec.NodeName, sanitizeLabelName(string(resourceName)), string(constant.UnitByte)},
+									Value:       float64(val.Value()),
+								})
+							}
+							if isExtendedResourceName(resourceName) {
+								ms = append(ms, &metric.Metric{
+									LabelValues: []string{t.lifecycle, "", t.name, p.Spec.NodeName, sanitizeLabelName(string(resourceName)), string(constant.UnitInteger)},
+									Value:       float64(val.Value()),
+								})
+							}
+						}
+					}
+				}
+
+				for _, metric := range ms {
+					metric.LabelKeys = []string{"lifecycle", "container", "type", "node", "resource", "unit"}
```

Comment:
Why are you adding a container label in the first place? If this is pod level metric we should not care about container level.